routing/endpointregistry.go

```diff
@@ -32,19 +34,18 @@ func (e *entry) InflightRequests() int64 {
 }
 
 type EndpointRegistry struct {
-	lastSeen        map[string]time.Time
 	lastSeenTimeout time.Duration
 	now             func() time.Time
 
-	mu sync.Mutex
-
-	data map[string]*entry
+	// map[string]*entry
+	data sync.Map
 }
 
 var _ PostProcessor = &EndpointRegistry{}
 
 type RegistryOptions struct {
 	LastSeenTimeout time.Duration
+	Detected        map[string]time.Time
```

Comment:
This additional option is needed to make `func (r *EndpointRegistry) setDetectedTime(key string, detected time.Time)` non-exported and still have tests working.